client/transport/src/web_sys.rs

```diff
@@ -0,0 +1,124 @@
+use std::time::Duration;
+
+use anyhow::anyhow;
+use futures_channel::{mpsc, oneshot};
+use futures_timer::Delay;
+use futures_util::future::{self, Either};
+use futures_util::StreamExt;
+use jsonrpsee_core::client::{TransportReceiverT, TransportSenderT};
+use jsonrpsee_core::{async_trait, Error};
+use wasm_bindgen::closure::Closure;
+use wasm_bindgen::{JsCast, JsValue};
+use web_sys::{CloseEvent, MessageEvent, WebSocket};
+
+#[derive(Debug)]
+enum WebSocketMessage {
+	Data(String),
+	Close,
+}
+
+/// Sender.
+#[derive(Debug)]
+pub struct Sender(mpsc::UnboundedSender<WebSocketMessage>);
```

Comment:
https://github.com/rustwasm/wasm-bindgen/pull/955

The `JsValue` is !Send and I had to use another channel here instead of having `WebSocket` here directly.